android-components/components/support/utils/src/test/java/mozilla/components/support/utils/DownloadUtilsTest.kt

```diff
@@ -128,6 +140,11 @@ class DownloadUtilsTest {
         assertEquals("file.data", DownloadUtils.guessFileName(null, null, "http://example.com/file.data", "application/octet-stream"))
         assertEquals("file.data", DownloadUtils.guessFileName(null, null, "http://example.com/file.data", "binary/octet-stream"))
         assertEquals("file.data", DownloadUtils.guessFileName(null, null, "http://example.com/file.data", "application/unknown"))
+
+        assertEquals("file.jpg", DownloadUtils.guessFileName(null, null, "http://example.com/file.zip", "image/jpeg"))
+        // Should not change to file.dll
+        assertEquals("file.exe", DownloadUtils.guessFileName(null, null, "http://example.com/file.exe", "application/x-msdos-program"))
```

Comment:
Should we also have a test that demonstrates that `.exe` will be correctly converted to? Something like:
`assertEquals("file.exe", DownloadUtils.guessFileName(null, null, "http://example.com/file.bin", "application/x-msdos-program"))`

Reply:
Sure, added!